bosh/ruby-app-release/unit-tests/spec/helpers.rb

```diff
@@ -0,0 +1,48 @@
+def check_block_formations(rendered_template)
+
+  line_index = 0
+  tokens = Array.new #acts like a stack
+  exec_line = rendered_template.each_line do |line|
+    if line.include? "case"
+      tokens.push("case")
+    elsif line.match? (/.+\)/)
+      tokens.push("open")
+    elsif line.include? ";;"
+      tokens.push("close")
+    end
+  end
+
+  case_token = tokens.shift
+  return false unless case_token == "case"
+  
+  remaining_tokens_count = tokens.length
+
+  next_token = "close"
+
+  for i in 1..remaining_tokens_count do
+    t = tokens.pop
+
+    return false unless t == next_token
+
+    if next_token == "close"
+      next_token = "open"
+    elsif
+      next_token = "close"
+    end
+  end
```

Comment:
```suggestion
  remaining_tokens_count.times do |token|
    return false unless token == next_token

    next_token == 'close' ? next_token = 'open' : next_token = 'close'
  end
```

Reply:
This won't work. `times` returns an integer index, whereas I need the item itself (i.e. `token`). 

The point here is that I am removing things from the array in LIFO order, I am not just iterating over it. Therefore I cannot do this with enumerables.